### ISSUE TITLE
Declare Fields with Nested Records of types defined in same Splice

### ISSUE THREAD

**Issue by jjlehner** [ekmett/lens#1032]
`declareFields [d|
  data A = A {  aSubB :: B }
  data B = B { bB :: Int }
 |]
`

At the moment a splice like this does not work with "error B_0 is not in scope at a reify". Could there be possible solutions to fix this without having to split it into two declaration groups?

**Comment by RyanGlScott** [comment]
I agree that this is unfortunate. I believe the reason this happens is because we use `reify` here to check if a field contains type families:

https://github.com/ekmett/lens/blob/edc14658d2bbb4ba981668a3f93b36547f43c01b/src/Control/Lens/Internal/FieldTH.hs#L434-L437

If we attempt to `reify` a field that hasn't been declared yet (e.g., `B`), then the error message above is the result. But we needn't fail here: we could just as well use [`recover`](https://hackage.haskell.org/package/template-haskell-2.20.0.0/docs/Language-Haskell-TH.html#v:recover) to catch the reification failure and then conservatively assume that the type doesn't contain type families, which is a reasonable assumption most of the time.
